tests/Unit/InvoiceTest.php

```diff
@@ -307,4 +307,66 @@ public function test_it_can_determine_if_reverse_charge_applies()
 
         $this->assertTrue($invoice->reverseChargeApplies());
     }
+    public function test_it_can_determine_if_is_deleted(): void
+    {
+        $customerId = 'foo';
+
+        $stripeInvoice = new StripeInvoice();
+
+        $stripeInvoice->customer = $customerId;
+        $stripeInvoice->status = StripeInvoice::STATUS_DELETED;
+
+        $user = new User();
+
+        $user->stripe_id = $customerId;
+
+        $invoice = new Invoice($user, $stripeInvoice);
+
+        $this->assertTrue($invoice->isDeleted());
+    }
+
+    /**
+     * @dataProvider provideInvoiceStatusNotDeleted
+     */
+    public function test_it_can_determine_if_is_not_deleted(string $invoiceStatus): void
+    {
+        $customerId = 'foo';
+
+        $stripeInvoice = new StripeInvoice();
+
+        $stripeInvoice->customer = $customerId;
+        $stripeInvoice->status = $invoiceStatus;
+
+        $user = new User();
+
+        $user->stripe_id = $customerId;
+
+        $invoice = new Invoice($user, $stripeInvoice);
+
+        $this->assertFalse($invoice->isDeleted());
```

Comment:
I don't know if anyone uses this method, but in the light of #1556, it either

- should always return `false`
- be deprecated and eventually removed